agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/localstorage/LocalFileLoader.java

```diff
@@ -92,18 +95,23 @@ PersistedFile loadTelemetriesFromDisk() {
       return null;
     }
 
-    byte[] result;
-    try {
-      // TODO (trask) optimization: read this directly into ByteBuffer(s)
-      result = Files.readAllBytes(tempFile.toPath());
+    ByteBuffer ikeyByteBuffer;
+    ByteBuffer result;
+    try (FileInputStream fileInputStream = new FileInputStream(tempFile);
+        FileChannel channel = fileInputStream.getChannel()) {
+      ikeyByteBuffer = ByteBuffer.allocate(36);
+      result = ByteBuffer.allocate((int) (tempFile.length() - 36));
+      channel.read(ikeyByteBuffer);
+      channel.read(result);
+      result.flip();
```

Comment:
should not ignore return value from channel.read(), generally have to loop since read can return before fully read

I don't think using ByteBuffer here helps, and would be simpler to read bytes directly from FileInputStream

Reply:
Updated.